# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/wr/RenderRootTypes.h
# Commit: c09a51622e98
# Full Hash: c09a51622e988e81077c75e3aaa5a49df66284b3
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2019-03-22 16:44:08
# Regressor Bug: 1441308
# File Overlap Count: 2
# Description:
#   Bug 1441308 - Core renderroot splitting changes r=kats,sotaro
#   
#   This is a large patch that contains all of the core changes for
#   renderroot splitting.
#   
# ==============================================================================

diff -r 737807563dd5 -r c09a51622e98 gfx/layers/wr/RenderRootTypes.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/gfx/layers/wr/RenderRootTypes.h	Fri Mar 22 03:15:14 2019 +0000
@@ -0,0 +1,69 @@
+/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set ts=8 sts=2 et sw=2 tw=80: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef GFX_RENDERROOTTYPES_H
+#define GFX_RENDERROOTTYPES_H
+
+#include "mozilla/webrender/WebRenderAPI.h"
+#include "mozilla/webrender/WebRenderTypes.h"
+#include "mozilla/layers/WebRenderMessages.h"
+#include "mozilla/layers/WebRenderScrollData.h"
+
+namespace mozilla {
+
+namespace layers {
+
+struct RenderRootDisplayListData {
+  wr::RenderRoot mRenderRoot;
+  gfx::IntRect mRect;
+  nsTArray<WebRenderParentCommand> mCommands;
+  wr::LayoutSize mContentSize;
+  mozilla::ipc::ByteBuf mDL;
+  wr::BuiltDisplayListDescriptor mDLDesc;
+  nsTArray<OpUpdateResource> mResourceUpdates;
+  nsTArray<RefCountedShmem> mSmallShmems;
+  nsTArray<mozilla::ipc::Shmem> mLargeShmems;
+  WebRenderScrollData mScrollData;
+};
+
+struct RenderRootUpdates {
+  wr::RenderRoot mRenderRoot;
+  nsTArray<WebRenderParentCommand> mCommands;
+  nsTArray<OpUpdateResource> mResourceUpdates;
+  nsTArray<RefCountedShmem> mSmallShmems;
+  nsTArray<mozilla::ipc::Shmem> mLargeShmems;
+  ScrollUpdatesMap mScrollUpdates;
+};
+
+}  // namespace layers
+
+namespace ipc {
+
+template <>
+struct IPDLParamTraits<mozilla::layers::RenderRootDisplayListData> {
+  typedef mozilla::layers::RenderRootDisplayListData paramType;
+
+  static void Write(IPC::Message* aMsg, IProtocol* aActor, paramType&& aParam);
+
+  static bool Read(const IPC::Message* aMsg, PickleIterator* aIter,
+                   IProtocol* aActor, paramType* aResult);
+};
+
+template <>
+struct IPDLParamTraits<mozilla::layers::RenderRootUpdates> {
+  typedef mozilla::layers::RenderRootUpdates paramType;
+
+  static void Write(IPC::Message* aMsg, IProtocol* aActor, paramType&& aParam);
+
+  static bool Read(const IPC::Message* aMsg, PickleIterator* aIter,
+                   IProtocol* aActor, paramType* aResult);
+};
+
+}  // namespace ipc
+}  // namespace mozilla
+
+
+#endif /* GFX_RENDERROOTTYPES_H */